php/libraries/Database.class.inc

```diff
@@ -165,11 +171,18 @@ class Database
 
         }
 
-        $this->_PDO = new PDO(
-            "mysql:host=$host;dbname=$database;charset=UTF8",
-            $username,
-            $password
-        );
+        try {
+            $this->_PDO = new PDO(
+                "mysql:host=$host;dbname=$database;charset=UTF8",
+                $username,
+                $password
+            );
+        } catch (PDOException $e) {
+            // This exception will not be included in the error log as it will
+            // likely include credentials.
+            error_log("Bad credentials or couldn't connect to database host.");
+            return false;
```

Comment:
This is changing the behavior of the connect function.
It would be "better" to thrown a \DatabaseException with a sanitized message.